DTS-STN/Alpha-Site
# Feedback api

# Description

[Feedback api endpoint](https://trello.com/c/Mt7tIKJY)

This is the api endpoint for the Feedback component that will be used in the projects. A POST request is made to `/api/feedback` with a body consisting of 
```
{
  project: "",
  pageUrl: "",
  feedback: ""
}
```
Validation will fail if any of these are missing from the body. And the `pageUrl` needs to be a URL.

On successful submit, an email will be sent to whatever you have in `NEXT_PUBLIC_NOTIFY_REPORT_A_PROBLEM_EMAIL`. Side note - it would be nice to change this env name, but that's unfortunately quite a bit of work across a few platforms.

## Acceptance Criteria

A successful POST to `/api/feedback` with the required data in the body sends an email to `NEXT_PUBLIC_NOTIFY_REPORT_A_PROBLEM_EMAIL`

## Test Instructions

1. Set your NEXT_PUBLIC_NOTIFY_REPORT_A_PROBLEM_EMAIL to something you have access to
2. Set USER_FEEDBACK_TEMPLATE_ID (ask me for the id, I don't want to post it here)
3. Run your app
3. Open Postman and create a post request to `localhost:3000/api/feedback`
4. Under the body section add the data that's required
5. Hit send and if you did it correctly, you should get an email


## Checklist


## Product and Sprint Backlog

[Trello](https://trello.com/b/ZqWtJSyh/alpha-site-board)
